package maidsafe.vault.protobuf;
import "maidsafe/vault/maid_account_holder/maid_account.proto";
option optimize_for = LITE_RUNTIME;

message Sync {
  required int32 action = 1;
  required bytes sync_message = 2;
}

message ArchiveFile {
  required FileId file_id = 1;
  required bytes contents = 2;
}

message GetArchiveFiles {
  repeated FileId file_hash_requested = 1;
}

message SyncInfo {
  required bytes maid_name = 1;
  required bytes maid_account = 2;
  optional bytes accumulator_entries = 3;
}

message SyncInfoResponse {
  required bytes node_id = 1;
  optional GetArchiveFiles file_hash_requests = 2;
}

message SyncArchiveFiles {
  required bytes maid_account = 1;
  repeated ArchiveFile files = 2;
}

message SyncArchiveFilesResponse {
  optional GetArchiveFiles file_hash_requests = 1;
}
